src/api/firebase.js

```diff
@@ -169,39 +169,87 @@ export async function shareList(listPath, currentUserId, recipientEmail) {
  */
+
 export async function addItem(listPath, { itemName, daysUntilNextPurchase }) {
 	const listCollectionRef = collection(db, listPath, 'items');
-
 	const newDocRef = doc(listCollectionRef);
 
 	return setDoc(newDocRef, {
 		dateCreated: new Date(),
-		// NOTE: This is null because the item has just been created.
-		// We'll use updateItem to put a Date here when the item is purchased!
 		dateLastPurchased: null,
 		dateNextPurchased: getFutureDate(daysUntilNextPurchase),
 		name: itemName,
 		totalPurchases: 0,
 	});
 }
 
+/**
+ * Updates the specified item in the shopping list with the new purchase information.
+ * @param {string} listPath - Path of the shopping list in Firestore.
+ * @param {string} itemId - ID of the item to be updated.
+ * @param {Timestamp} dateLastPurchased - Timestamp of the last purchase date.
+ * @param {number} nextPurchaseEstimate - Estimated number of days until the next purchase.
+ * @returns {Promise} A promise that resolves when the item is successfully updated.
+ */
 export async function updateItem(
 	listPath,
 	itemId,
+	todaysDate,
 	dateLastPurchased,
+	dateNextPurchased,
 	nextPurchaseEstimate,
 ) {
 	const listCollectionRef = collection(db, listPath, 'items');
 
 	const itemDocRef = doc(listCollectionRef, itemId);
 
 	return updateDoc(itemDocRef, {
-		dateLastPurchased,
+		previousNextPurchased: dateNextPurchased,
+		previousLastPurchased: dateLastPurchased,
+		dateLastPurchased: todaysDate,
 		dateNextPurchased: getFutureDate(nextPurchaseEstimate),
 		totalPurchases: increment(1),
 	});
 }
 
+/**
+ * Removes the last purchase information from the specified item in the shopping list.
+ * If the item has no previous purchases, the function resolves without making any changes.
+ * @param {string} listPath - Path of the shopping list in Firestore.
+ * @param {string} itemId - ID of the item to be unchecked.
+ * @returns {Promise} A promise that resolves when the item is successfully unchecked or if there are no changes needed.
+ */
+
+export async function uncheckItem(
+	listPath,
+	itemId,
+	previousLastPurchased,
+	previousNextPurchased,
+) {
+	// Create a reference to the item document in the specified shopping list
+	const listCollectionRef = collection(db, listPath, 'items');
+	const itemDocRef = doc(listCollectionRef, itemId);
+
+	// Retrieve the item document data
+	const itemDoc = await getDoc(itemDocRef);
+	const itemData = itemDoc.data();
+
+	const newTotalPurchases = itemData.totalPurchases - 1;
+
+	// Check if the item has a previous purchase
+	if (itemData.dateLastPurchased) {
+		// Update the item document with the new information
+		updateDoc(itemDocRef, {
+			dateLastPurchased: previousLastPurchased,
+			dateNextPurchased: previousNextPurchased,
+			totalPurchases: newTotalPurchases,
+		});
+	} else {
+		// If the item has no previous purchases, resolve the promise without making any changes
+		return Promise.resolve();
```

Comment:
I don't believe that we need `Promise.resolve` here. The`unCheckltem` function is not returning any specific value, so it can directly return without it. The function is` async` so it already returns a promise. Returning `Promise.resolve` explicitly creates a new Promise that resolves with the given value. Here is the link to the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve) -- my explanation is only partial.

Reply:
To add to @StefieCaff 's comment above I'm wondering in general the purpose of returning `Promise.resolve()` here